### PR TITLE
Always use h1 for first heading on a page

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Related to https://github.com/elastic/docs/pull/2768 (specifically https://github.com/elastic/docs/pull/2768#issuecomment-2138009956)
Fixes https://github.com/elastic/docs/issues/1614

~~I _think_ this should force the first heading on every page to be an `h1`. However, this fix seems to easy so I'm not sure if I'm misunderstanding the total impact of this change. Opening this PR to see if the PR preview builds and what it looks like before fixing tests.~~ Switched to a different approach detailed in https://github.com/elastic/docs/pull/3009#issuecomment-2159328351 below.

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
It looks like this "fix" sometimes results in pages with multiple `h1` headings on (I _think_) pages that don't use `[discrete]` on headings that should be on the same page so I'm not sure it's much of a fix after all... I'm not sure if I should dig in further or rely on just styling the headings appropriately like we're doing in #2768.

**Comment by bmorelli25** [comment]
Can you provide a double H1 example? I'm having trouble finding one.

**Comment by colleenmcginnis** [comment]
Here are all of them in the 8.13 version of the Observability guide:

* [`html/en/observability/8.13/apm-redis-output.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-redis-output.html)
* [`html/en/observability/8.13/apm-privileges-rum-source-map.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-privileges-rum-source-map.html)
* [`html/en/observability/8.13/apm-privileges-agent-central-config.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-privileges-agent-central-config.html)
* [`html/en/observability/8.13/apm-logstash-output.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-logstash-output.html)
* [`html/en/observability/8.13/apm-kafka-output.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-kafka-output.html)
* [`html/en/observability/8.13/apm-elasticsearch-output.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-elasticsearch-output.html)
* [`html/en/observability/8.13/apm-console-output.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-console-output.html)
* [`html/en/observability/8.13/apm-configure-tail-based-sampling.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-configure-tail-based-sampling.html)
* [`html/en/observability/8.13/apm-configure-head-based-sampling.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-configure-head-based-sampling.html)
* [`html/en/observability/8.13/apm-configure-cloud-id.html`](https://github.com/elastic/built-docs/tree/docs_bk_3009/html/en/observability/8.13/apm-configure-cloud-id.html)

So there are not too many in Observability docs, but in bigger/older books there are many more. For example, in v8.13 of the Elasticsearch guide 843 out of 1815 HTML files contain more than one `h1`.

**Comment by bmorelli25** [comment]
Discussed in metrics mtg this morning. @KOTungseth will ask Katy on Friday if no H1 is better or worse than multiple H1s.

**Comment by colleenmcginnis** [comment]
Ok I tried a different approach here: trying to use the built-in AsciiDoc document header. It looks like we set the `noheader` option to `true`. I'm not sure why or if there will be unintended consequences by turning this off, but I haven't come across a dealbreaker... yet.

It's not as simple as changing that option. This PR would also:

* **Move the breadcrumbs**. If we use the document header with its h1, the current placement of the breadcrumbs would be _below_ the title. To address this, we can create a placeholder `div` for the breadcrumbs when we create the header, and then replace the placeholder when we go through the chunk logic. (Note: This is breaking the tests. We need to change where to expect the breadcrumbs.)
* **Hide the first heading in `div#content`**. When we use the document header, the title is still also added in the body of the doc (specifically in `div#content`), but we can hide that heading with CSS.
* **Still hack incremental heading styles**. The logic in this PR would ensure two things: (1) the first heading on each page is an h1 and (2) that will be the only h1 on the page. _However_, it doesn't guarantee that the headings in the main body of the doc will be incremental, but we can style them appropriately at run time.

**Comment by bmorelli25** [comment]
Clicked around a bit.
- The [docs landing page](https://docs_bk_3009.docs-preview.app.elstc.co/guide/index.html) is a bit broken with this new update.
- It looks like OTP highlighting doesn't work, BUT, I love that the OTP exists on every page now and is even expanded on a lot of pages due to every page starting with H1.

**Comment by colleenmcginnis** [comment]
>Clicked around a bit.
>
>* The [docs landing page](https://docs_bk_3009.docs-preview.app.elstc.co/guide/index.html) is a bit broken with this new update.
>* It looks like OTP highlighting doesn't work, BUT, I love that the OTP exists on every page now and is even expanded on a lot of pages due to every page starting with H1.

**Comment by bmorelli25** [comment]
Awesome!

File: integtest/spec/single_book_spec.rb [hunk 1]
```diff
@@ -314,133 +314,7 @@
...
-        xpack_tag_context true, true, true, true, false
-      end
-    end
-    context 'when hiding xpack tags' do
-      context 'when the xpack role is on everything' do
-        xpack_tag_context true, true, true, true, true
-      end
-    end
-  end
+  # Do we still need the xpack tag tests?
```
Comment (bmorelli25) [inline integtest/spec/single_book_spec.rb:317]: I guess not...?

Comment (bmorelli25) [reply integtest/spec/single_book_spec.rb:317]: I vote to remove. https://github.com/elastic/docs/pull/2213

**Review by bmorelli25 — APPROVED** [review]
This looks great. Nice fix for that 🐛 in 34a16c6. I've clicked around as much as I can, and I'm not seeing any new 🐛s.

🐦

**Pull request merged**